homeassistant/components/mqtt/client.py

```diff
@@ -797,10 +797,12 @@ def _mqtt_handle_message(self, msg: mqtt.MQTTMessage) -> None:
         for subscription in subscriptions:
             init_status = self._retained_init.setdefault(subscription, set())
             # skip already initialized subscriptions
-            if msg.retain and msg.topic in init_status:
-                continue
-            # remember the subscription had an initial payload
-            self._retained_init[subscription].add(msg.topic)
+            if msg.topic in init_status:
+                if msg.retain:
```

Comment:
Why not make this the outer check?

In both the if and elif, you're only doing something if msg.retain, so instead:
```
if msg.retain:
    init_status = self._retained_init.setdefault(subscription, set())
    if msg.topic in init_status:
        continue
    else
        self.retained_init[subscription].add(msg.topic)
``

Reply:
> if msg.retain:

Better indeed, we can even remove the else statement.